scripts/testTKG/test_platformRequirements.py

```diff
@@ -16,105 +16,134 @@
+	else:
+		passed.add('test_arch_nonx86_0')
+		skipped.add('test_arch_x86_0')
+
+	if '64' in spec:
+		passed.add('test_bits_64_0')
+	else:
+		skipped.add('test_bits_64_0')
+
+	if 'osx' in spec:
+		passed.add('test_os_osx_0')
+	else:
+		skipped.add('test_os_osx_0')
+
+	if 'osx_x86-64' in spec:
+		passed.add('test_osx_x86-64_0')
+	else:
+		skipped.add('test_osx_x86-64_0')
+
+	if 'x86' in spec or '390' in spec:
+		passed.add('test_arch_x86_390_0')
+	else:
+		skipped.add('test_arch_x86_390_0')
+
+	if 'osx_x86-64' in spec or ('win_x86' in spec and 'win_x86-64' not in spec) or 'aix_ppc-64' in spec:
+		passed.add('test_osx_x86-64_win_x86_aix_ppc-64_0')
+	else:
+		skipped.add('test_osx_x86-64_win_x86_aix_ppc-64_0')
+
+	os_label = re.search(r"set OS_LABEL to (.*)", stdout)
+	if os_label is not None:
+		os_label = os_label.group(1)
+		label_str = os_label.split(".")
+		# os_label example: ubuntu.22
+		try:
+			ver = int(label_str[1],10)
+			if label_str[0] == "ubuntu":
+				if ver >= 22:
+					passed.add('test_os_linux_ubuntu22_0')
+				if ver >= 20:
+					passed.add('test_os_linux_ubuntu20plus_0')
+					passed.add('test_os_linux_ubuntu20plus_rhel8plus_0')
+				if ver < 20:
+					passed.add('test_not_os_linux_ubuntu20plus_0')
+
+			if label_str[0] == "rhel":
+				if ver >= 8:
+					passed.add('test_os_linux_ubuntu20plus_rhel8plus_0')
+		except ValueError as ve:
+			print ("warning: os version value failed to convert to an integer")
+			passed.add('test_not_os_linux_ubuntu20plus_0')
+	else:
+		passed.add('test_not_os_linux_ubuntu20plus_0')
+
+	if 'test_os_linux_ubuntu20plus_0' not in passed:
+		skipped.add('test_os_linux_ubuntu20plus_0')
+
+	if 'test_os_linux_ubuntu22_0' not in passed:
+		skipped.add('test_os_linux_ubuntu22_0')
+
+	if 'test_os_linux_ubuntu20plus_rhel8plus_0' not in passed:
+		skipped.add('test_os_linux_ubuntu20plus_rhel8plus_0')
+
+	if 'test_not_os_linux_ubuntu20plus_0' not in passed:
+		skipped.add('test_not_os_linux_ubuntu20plus_0')
+
+	micro_arch = re.search(r"set MICROARCH to (.*)", stdout)
+	print("Output:", stdout)
+	if micro_arch is not None:
+		micro_arch = micro_arch.group(1)
+		label_str = os_label.split(".")
+		# micro_arch example: skylake
+		try:
+			if label_str[0] == "skylake":
+				passed.add('test_microarchitecture_detection_skylake_0')
+			elif label_str[0] == "z15":
+				passed.add('test_microarchitecture_detection_z15')
+				passed.add('test_microarchitecture_detection_z13_and_above_0')
+			elif label_str[0] == "z14":
+				passed.add('test_microarchitecture_detection_z14')
+				passed.add('test_microarchitecture_detection_z13_and_above_0')
+			elif label_str[0] == "z13s":
+				passed.add('test_microarchitecture_detection_z13s_0')
+				passed.add('test_microarchitecture_detection_z13_and_above_0')
+			elif label_str[0] == "z13+":
+				passed.add('test_microarchitecture_detection_z13_and_above_0')
+			elif label_str[0] == "z13":
+				passed.add('test_microarchitecture_detection_z13_0')
+			else:
```

Comment:
This code needs to be more robust. Please use numeric comparison. We also need to consider z14+, z15+, etc

if version >= 13, then it is z13+
if version >= 14, then it is z14+
...